docs/DeveloperGuide.md

```diff
@@ -411,18 +452,47 @@ The tradeoffs for this approach is examined below:
       * This may lead to unexpected behaviours, such as whether properties linked to an owner should be deleted when
       the owner is deleted.
 
-### Filtering properties by price range
+### Filtering buyers and properties: Overall structure
 
-The `Properties` list is filtered using a predicate, `filterPropsByPricePredicate`. This predicate checks if the
-property's price falls within a specified price range.
+In order to filter `Buyers` and `Properties`, a `Predicate` needs to be passed into the `ObservableArrayList` that stores 
+references to these objects and displays them on the user's screen. These predicates can differ in the conditions that are
+being tested, consequently, they might give different outputs when applied to a given list.
 
-The structure for executing a `filterprops` command follows the flow as mentioned in the “Logic component” section of
-this guide.
+#### Design Considerations
+In order to allow for multiple-condition filtering, that is, the composition of multiple filter predicates, an abstract 
+`AbstractFilterXYZPredicate` class was created to employ polymorphic behaviour, where XYZ represents the entry type that
+we are working with, for example `AbstractFilterBuyerPredicate` or `AbstractFilterPropsPredicate`. 
```

Comment:
I just realised our naming for the predicates are inconsistent, Buyer is singular while Props is plural. We should just use plural for Buyer too haha